src/cloud/components/Comments/ThreadItem.tsx

```diff
@@ -43,80 +42,132 @@ function ThreadItem({
   addReaction,
   removeReaction,
   user,
-}: ThreadListItemProps) {
-  const { translate } = useI18n()
+  onCommentDelete,
+}: ThreadListItemProps) => {
   const [editing, setEditing] = useState(false)
+  const { messageBox } = useDialog()
+  const { translate } = useI18n()
 
-  const [threadComments, setThreadComments] = useState<Comment[] | null>(null)
   const [showingContextMenu, setShowingContextMenu] = useState<boolean>(false)
-  const reloadComments = useCallback(async () => {
-    const comments = await listThreadComments({ id: thread.id })
-    const sortedComments = sortBy(prop('createdAt'), comments)
-    setThreadComments(sortedComments)
-  }, [thread.id])
-
-  useEffect(() => {
-    reloadComments()
-  }, [reloadComments, thread.id])
-
-  const showReplyForm = useCallback(() => {
-    if (threadComments == null || threadComments.length > 1) {
-      return
-    }
 
+  const showContextMenu = useCallback(() => {
     setShowingContextMenu(true)
-  }, [threadComments])
+  }, [])
 
-  const hideReplyForm = useCallback(() => {
+  const hideContextMenu = useCallback(() => {
     setShowingContextMenu(false)
   }, [])
 
-  const threadOpenerComment = useMemo(() => {
-    if (threadComments == null || threadComments.length == 0) {
-      return
-    }
-    return threadComments[0]
-  }, [threadComments])
-
-  const threadOpenedUser = useMemo(() => {
-    if (threadComments == null || threadComments.length == 0 || user == null) {
-      return
-    }
-
-    return threadComments[0].user
-  }, [threadComments, user])
-
   const submitComment = useCallback(
     async (message: string) => {
-      if (threadOpenerComment == null) {
+      if (thread.initialComment == null) {
         return
       }
-      await updateComment(threadOpenerComment, message)
+      await updateComment(thread.initialComment, message)
       setEditing(false)
-      reloadComments()
     },
-    [reloadComments, threadOpenerComment, updateComment]
+    [thread.initialComment, updateComment]
   )
 
+  const replyCount = useMemo(() => {
+    return thread.initialComment == null
+      ? thread.commentCount
+      : thread.commentCount - 1
+  }, [thread.commentCount, thread.initialComment])
+
   const onThreadDelete = useCallback(
-    (thread) => {
-      onDelete(thread)
-      reloadComments()
+    async (thread: Thread) => {
+      if (thread == null) {
+        return
+      }
+
+      if (thread.initialComment == null || replyCount === 0) {
```

Comment:
I am a bit confused by this logic here.
so a user will still be to completely delete threads that have replies?
They just need to remove their initial comment and then the thread itself?  If we want to mimic slack it is not correct.

it does not make sense to me. It should be back end logic.

in the front end: always send `deleteComment`
in the backend: delete comment, if thread affiliated has no more comments, remove thread.

Reply:
The onThreadDelete is fired when user can click on trash button, but the button itself is shown only if there are no replies, so no, user cannot do this afaik.

User needs to have 0 replies nevertheless of inital comment being removed